upgrade-order-processing.md

```diff
@@ -0,0 +1,357 @@
+# Upgrade Order Processing
+
+## General changes that have to be known
+
+This article covers the changes to the order processing system implemented in the latest release.
+These changes aim to enhance functionality, improve maintainability, and make implementing the changes easier.
+
+### What has changed?
+
+Introducing new middleware components to the order processing system instead of previously used order preview calculation.
```

Comment:
A link to the article in the docs would fit nicely here in this section somewhere for sure :slightly_smiling_face: